src/util/irep.h

```diff
@@ -82,8 +82,77 @@ inline const std::string &name2string(const irep_namet &n)
 class irept;
 const irept &get_nil_irep();
 
-/*! \brief Base class for tree-like data structures with sharing
-*/
+/// \brief Base class for tree-like data structures with sharing
+///
+/// There are a large number of kinds of tree structured or tree-like data in
+/// CPROVER. `irept` provides a single, unified representation for all of
+/// these, allowing structure sharing and reference counting of data. As
+/// such `irept` is the basic unit of data in CPROVER.  Each `irept`
+/// contains[^1] a basic unit of data (of type `dt`) which contains four
+/// things:
+///
+/// * `data`:   A string[^2], which is returned when the `id()` function is
+///   used.
+///
+/// * `named_sub`:   A map from `irep_namet` (a string) to `irept`. This
+///   is used for named children, i.e.  subexpressions, parameters, etc.
+///
+/// * `comments`:   Another map from `irep_namet` to `irept` which is used
+///   for annotations and other ‘non-semantic’ information
+///
+/// * `sub`:   A vector of `irept` which is used to store ordered but
+///   unnamed children.
+///
+/// The `irept::pretty` function outputs the contents of an `irept` directly
+/// and can be used to understand and debug problems with `irept`s.
+///
+/// On their own `irept`s do not “mean” anything; they are effectively
+/// generic tree nodes. Their interpretation depends on the contents of
+/// result of the `id` function (the `data`) field. `util/irep_ids.txt`
+/// contains the complete list of `id` values. During the build process it
```

Comment:
Nah, ireps that use other strings (e.g. `irept("Hello world")` are not common but are definitely present in the codebase